http-endpoint/src/basic_auth.rs

```diff
@@ -0,0 +1,59 @@
+use actix_web::dev::ServiceRequest;
+use actix_web::http::header;
+use actix_web::Error;
+
+use actix_web_httpauth::extractors::basic::{Config, BasicAuth};
+
+use actix_web::client::Client;
+use awc::http::StatusCode;
+use log;
+use actix_web_httpauth::extractors::AuthenticationError;
+
+const AUTH_SERVICE_URL: &str = "AUTH_SERVICE_URL";
+
+pub async fn basic_validator(
+    req: ServiceRequest,
+    cred: BasicAuth,
+) -> Result<ServiceRequest, Error> {
+
+    //TODO : get this when initializing the app instead of pulling it each time
+    let auth_service_url = std::env::var(AUTH_SERVICE_URL)
+        .expect("AUTH_SERVICE_URL must be set");
+
+    let config = req
+        .app_data::<Config>()
+        .map(|data| data.clone())
+        .unwrap_or_else(Default::default);
+
+    let url = format!("http://{}/auth", auth_service_url);
+
+    // the unwrap is safe here because the BasicAuth extractor already check the header.
+    // We fetch the encoded header to avoid decoding then re-encoding
+    let encoded_basic_header = req.headers().get(header::AUTHORIZATION).unwrap();
+        // match {
+        //     Some(h) => h,
+        //     None => return Err(ErrorBadRequest(Err("Missing Authorization header"))),
+        // };
+
```

Comment:
I wanted to do a proper check here but i was getting all sort of `this error does not implement actix::Error whatever Trait` so I resigned to use unwrap, as it should be safe at that point..

Reply:
So, if your error ("bad request") can be converted to the actix error, then you can to with e.g. with:

~~~rust
return Err(ErrorBadRequest(Err("Missing Authorization header")).into())
~~~

As you don't use `?`, this needs to be done explicitly.

You could also do something like:

~~~rust
let header = match header {
  Some(header) => Ok(header),
  None => Err(…),
}?;
~~~

Which should be the same as:

~~~rust
let header = header.ok_or_else(||MyErr{})?;
~~~

IIRC the `BasicAuth` parameter should also have this information.